Neos.Neos/Classes/Controller/Module/Administration/UsersController.php

```diff
@@ -368,9 +368,9 @@ public function updateAccountAction(Account $account, array $roleIdentifiers, ar
             $this->redirect('index');
         }
         if ($user === $this->currentUser) {
-            $roles = [];
+            $roles = array_map(fn (string $roleIdentifier) => $this->policyService->getRole($roleIdentifier), $roleIdentifiers);
             foreach ($roleIdentifiers as $roleIdentifier) {
-                $roles[$roleIdentifier] = $this->policyService->getRole($roleIdentifier);
+                $roles[] = $this->policyService->getRole($roleIdentifier);
```

Comment:
This is doubled, isn't it? Maybe caused by your PR split?

Reply:
Ah, I meant to revert this change. Thanks!